blenderproc/python/utility/Initializer.py

```diff
@@ -164,18 +123,25 @@ def set_default_parameters():
 
 
     @staticmethod
-    def _remove_all_data():
-        """ Remove all data blocks except opened scripts and the default scene. """
+    def _remove_all_data(remove_camera: bool = True):
+        """ Remove all data blocks except opened scripts, the default scene and the camera. 
+        
+        :param remove_camera: If True, also the default camera is removed.
+        """
         # Go through all attributes of bpy.data
         for collection in dir(bpy.data):
             data_structure = getattr(bpy.data, collection)
             # Check that it is a data collection
             if isinstance(data_structure, bpy.types.bpy_prop_collection) and hasattr(data_structure, "remove") and collection not in ["texts"]:
                 # Go over all entities in that collection
                 for block in data_structure:
-                    # Remove everything besides the default scene
-                    if not isinstance(block, bpy.types.Scene) or block.name != "Scene":
-                        data_structure.remove(block)
+                    # Skip the default scene
+                    if isinstance(block, bpy.types.Scene) and block.name == "Scene":
+                        continue
+                    # If desired, skip camera
+                    if not remove_camera and (isinstance(block, bpy.types.Object) or isinstance(block, bpy.types.Camera)) and block.name == "Camera":
```

Comment:
No, objects are only things that are placed in the scene graph, but there are a lot of other types of data blocks in blender, for example materials, lights, armatures, images, meshes etc. see https://docs.blender.org/api/current/bpy.types.BlendData.html

Its a bit confusing, because e.g. cameras consist of two data blocks. One `bpy.types.Object` block for positioning it in the world and one `bpy.types.Camera` block for handling all camera specific parameters.